client/homebrew/utils/versionHistory.js

```diff
@@ -43,74 +50,110 @@ function updateStoredBrew(brew, slot = 0) {
+		HISTORY_SAVE_DELAYS = global.config?.history?.HISTORY_SAVE_DELAYS;
+	}
+
 	archiveBrew.expireAt.setMinutes(archiveBrew.expireAt.getMinutes() + HISTORY_SAVE_DELAYS[slot]);
 
 	const key = getKeyBySlot(brew, slot);
-	localStorage.setItem(key, JSON.stringify(archiveBrew));
+
+	return [key, archiveBrew];
 }
 
 
-export function historyExists(brew){
-	return Object.keys(localStorage)
-		.some((key)=>{
-			return key.startsWith(`${HISTORY_PREFIX}-${brew.shareId}`);
-		});
+async function createHBStore(){
+	return await IDB.createStore(HB_DB, HB_STORE);
+}
+
+export async function historyCheck(brew){
+	if(!IDB) return false;
+	const historyExists = await IDB.keys(await createHBStore())
+		.then((keys)=>{
+			return [...keys].some((key)=>{
+				return key.startsWith(`${HISTORY_PREFIX}-${brew.shareId}`);
+			});
+		})
+		.catch(()=>{return false;});
+
+	return historyExists;
 }
 
-export function loadHistory(brew){
-	const history = {};
+export async function loadHistory(brew){
+	const DEFAULT_HISTORY_ITEM = { expireAt: '2000-01-01T00:00:00.000Z', shareId: brew.shareId, noData: true };
+
+	const historyKeys = [];
 
-	// Load data from local storage to History object
+	// Create array of all history keys
 	for (let i = 1; i <= HISTORY_SLOTS; i++){
-		history[i] = getVersionBySlot(brew, i);
+		historyKeys.push(getKeyBySlot(brew, i));
 	};
 
+	const history = [];
+	// Load all keys from IDB at once
+	await IDB.getMany(historyKeys, await createHBStore())
+				.then((dataArray)=>{
+					return dataArray.forEach((data)=>{
+						history.push(data ?? DEFAULT_HISTORY_ITEM);
+					});
+				})
+				.catch(()=>{
+					historyKeys.forEach(()=>{
+						history.push(DEFAULT_HISTORY_ITEM);
+					});
+				});
+
 	return history;
 }
 
-export function updateHistory(brew) {
-	const history = loadHistory(brew);
+export async function updateHistory(brew) {
+	const history = await loadHistory(brew);
 
 	// Walk each version position
-	for (let slot = HISTORY_SLOTS; slot > 0; slot--){
+	for (let slot = HISTORY_SLOTS - 1; slot >= 0; slot--){
 		const storedVersion = history[slot];
 
 		// If slot has expired, update all lower slots and break
 		if(new Date() >= new Date(storedVersion.expireAt)){
-			for (let updateSlot = slot - 1; updateSlot>0; updateSlot--){
+
+			// Create array of arrays : [ [key1, value1], [key2, value2], ..., [keyN, valueN] ]
+			// to pass to IDB.setMany
+			const historyUpdate = [];
+
+			for (let updateSlot = slot; updateSlot > 0; updateSlot--){
 				// Move data from updateSlot to updateSlot + 1
-				!history[updateSlot]?.noData &&	updateStoredBrew(history[updateSlot], updateSlot + 1);
+				if(!history[updateSlot - 1]?.noData) {
+					historyUpdate.push(parseBrewForStorage(history[updateSlot - 1], updateSlot + 1));
+				}
 			};
 
 			// Update the most recent brew
-			updateStoredBrew(brew, 1);
+			historyUpdate.push(parseBrewForStorage(brew, 1));
+
+			IDB.setMany(historyUpdate, await createHBStore());
```

Comment:
Does this need an await? I guess nothing else relies on the result here so it can be done asynchronously,

Reply:
Technically, no it doesn't.

Possibly it should have one, just for consistency.

Reply:
I've added await to this call and also the Garbage Collection IDB.delete function, so that the IDB calls are all consistently called with await.